enaml/testing/utils.py

```diff
@@ -0,0 +1,163 @@
+#------------------------------------------------------------------------------
+# Copyright (c) 2013-2017, Nucleic Development Team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+#------------------------------------------------------------------------------
+"""Generic utility functions for testing.
+
+"""
+from time import sleep
+from contextlib import contextmanager
+
+import enaml
+from ..application import timed_call
+from ..widgets.api import Window, Dialog
+with enaml.imports():
+    from ..stdlib.message_box import MessageBox
+
+
+def get_window(app, cls=Window):
+    """Convenience function running the event loop and returning the first
+    window found in the set of active windows.
+
+    Parameters
+    ----------
+    cls : type, optional
+        Type of the window which should be returned.
+
+    """
+    sleep(0.1)
+    app.process_events()
+    w_ = None
+    for w in Window.windows:
+        if isinstance(w, cls):
+            w_ = w
+            break
+
+    return w_
+
+
+def close_all_windows(app):
+    """Close all opened windows.
+
+    This should be used by all tests creating windows in a teardown step.
+
+    """
+    app.process_events()
+    sleep(0.1)
+    while Window.windows:
+        for window in list(Window.windows):
+            window.close()
+        app.process_events()
+        sleep(0.02)
```

Comment:
why sleep?

Reply:
In case any thread decides to create a new window after this was called but also because sometimes Qt handling (and os related actions) take some time to actually take effect, even if I am not sure why.